tests/driver/lint_driver.ts

```diff
@@ -0,0 +1,304 @@
+import Debug from "debug";
+import * as fs from "fs";
+import path from "path";
+import { SetupSettings } from "tests/driver";
+import { LandingState, TrunkVerb } from "tests/types";
+import { ARGS } from "tests/utils";
+import { tryParseLandingState } from "tests/utils/landing_state";
+import { getTrunkVersion } from "tests/utils/trunk_config";
+
+import { GenericTrunkDriver } from "./driver";
+
+const baseDebug = Debug("Driver");
+let testNum = 1;
+const linterTests = new Map<string, number>();
+
+const getDebugger = (linter?: string) => {
+  if (!linter) {
+    // If a linter is not provided, provide a counter for easy distinction
+    return baseDebug.extend(`test${testNum++}`);
+  }
+  const numLinterTests = linterTests.get(linter);
+  const newNum = (numLinterTests ?? 0) + 1;
+  linterTests.set(linter, newNum);
+  return baseDebug.extend(linter).extend(`${newNum}`);
+};
```

Comment:
Really, this should be its own util class or something. Up to you on if you want to do it now or later